Will not build under Ubuntu 21.04/riscv64

Using clang:
Ubuntu clang version 12.0.0-3ubuntu1~21.04.2
Target: riscv64-unknown-linux-gnu

clang -g -W -Wall   -c -o main.o main.c
clang -g -W -Wall   -c -o pdp10-opc.o pdp10-opc.c
pdp10-opc.c:396:47: warning: missing field 'ac_hint' initializer [-Wmissing-field-initializers]
  { "fix",      0247,   PDP10_BASIC,    PDP10_KA10_SAIL },
                                                        ^
1 warning generated.
clang -g -W -Wall   -c -o info.o info.c
info.c:573:4: warning: format specifies type 'unsigned int' but the argument has type 'long long' [-Wformat]
                 word & 0777777, (word >> 18) & 0777777);
                 ^~~~~~~~~~~~~~
info.c:573:20: warning: format specifies type 'unsigned int' but the argument has type 'long long' [-Wformat]
                 word & 0777777, (word >> 18) & 0777777);
                                 ^~~~~~~~~~~~~~~~~~~~~~
2 warnings generated.
clang -g -W -Wall   -c -o dis.o dis.c
clang -g -W -Wall   -c -o symbols.o symbols.c
clang -g -W -Wall   -c -o timing.o timing.c
clang -g -W -Wall   -c -o timing_ka10.o timing_ka10.c
clang -g -W -Wall   -c -o timing_ki10.o timing_ki10.c
clang -g -W -Wall   -c -o memory.o memory.c
clang -g -W -Wall   -c -o weenix.o weenix.c
clang -g -W -Wall   -c -o fasl-file.o fasl-file.c
fasl-file.c:52:7: error: expected expression
      char *p = string;
      ^
fasl-file.c:57:12: error: use of undeclared identifier 'p'
          *p++ = (data >> 29) & 0177;
           ^
fasl-file.c:58:12: error: use of undeclared identifier 'p'
          *p++ = (data >> 22) & 0177;
           ^
fasl-file.c:59:12: error: use of undeclared identifier 'p'
          *p++ = (data >> 15) & 0177;
           ^
fasl-file.c:60:12: error: use of undeclared identifier 'p'
          *p++ = (data >>  8) & 0177;
           ^
fasl-file.c:61:12: error: use of undeclared identifier 'p'
          *p++ = (data >>  1) & 0177;
           ^
6 errors generated.
make: *** [<builtin>: fasl-file.o] Error 1

Using GCC 10:
gcc -g -W -Wall   -c -o main.o main.c
gcc -g -W -Wall   -c -o pdp10-opc.o pdp10-opc.c
pdp10-opc.c:396:3: warning: missing initializer for field ‘ac_hint’ of ‘const struct pdp10_instruction’ [-Wmissing-field-initializers]
  396 |   { "fix", 0247, PDP10_BASIC, PDP10_KA10_SAIL },
      |   ^
In file included from pdp10-opc.c:19:
opcode/pdp10.h:126:7: note: ‘ac_hint’ declared here
  126 |   int ac_hint;
      |       ^~~~~~~
gcc -g -W -Wall   -c -o info.o info.c
info.c: In function ‘dec_info’:
info.c:572:35: warning: format ‘%o’ expects argument of type ‘unsigned int’, but argument 3 has type ‘word_t’ {aka ‘long long int’} [-Wformat=]
  572 |  fprintf (output_file, "DDT from %o to %o\n",
      |                                  ~^
      |                                   |
      |                                   unsigned int
      |                                  %llo
  573 |    word & 0777777, (word >> 18) & 0777777);
      |    ~~~~~~~~~~~~~~
      |         |
      |         word_t {aka long long int}
info.c:572:41: warning: format ‘%o’ expects argument of type ‘unsigned int’, but argument 4 has type ‘word_t’ {aka ‘long long int’} [-Wformat=]
  572 |  fprintf (output_file, "DDT from %o to %o\n",
      |                                        ~^
      |                                         |
      |                                         unsigned int
      |                                        %llo
  573 |    word & 0777777, (word >> 18) & 0777777);
      |                    ~~~~~~~~~~~~~~~~~~~~~~
      |                                 |
      |                                 word_t {aka long long int}
gcc -g -W -Wall   -c -o dis.o dis.c
gcc -g -W -Wall   -c -o symbols.o symbols.c
gcc -g -W -Wall   -c -o timing.o timing.c
gcc -g -W -Wall   -c -o timing_ka10.o timing_ka10.c
gcc -g -W -Wall   -c -o timing_ki10.o timing_ki10.c
gcc -g -W -Wall   -c -o memory.o memory.c
gcc -g -W -Wall   -c -o weenix.o weenix.c
gcc -g -W -Wall   -c -o fasl-file.o fasl-file.c
fasl-file.c: In function ‘read_atomtable’:
fasl-file.c:52:7: error: a label can only be part of a statement and a declaration is not a statement
   52 |       char *p = string;
      |       ^~~~
make: *** [<builtin>: fasl-file.o] Error 1

--- a/fasl-file.c
+++ b/fasl-file.c
@@ -43,13 +43,14 @@ read_atomtable (FILE *f, word_t header)
   char string[100];
   int length;
   word_t data;
+  char *p;
 
   //fprintf (stderr, "Atomtable type %llo:  ", (header >> 33) & 7);
   type[atom_index] = (header >> 33) & 7;
   switch (type[atom_index])
     {
     case 0: //Atom.
-      char *p = string;
+      p = string;
       length = header & 0777777;
       while (length--)
         {

--- a/info.c
+++ b/info.c
@@ -663,7 +663,6 @@ byte_size (int code, int *leftovers)
 void
 ntsddt_info (struct pdp10_memory *memory, int ddt)
 {
-  extern word_t get_word_at ();
   char name[7];
   word_t a, w, v;
   int i, syms;

--- a/info.c
+++ b/info.c
@@ -569,7 +569,7 @@ dec_info (struct pdp10_memory *memory,
 
       word = get_word_at (memory, JBREL);
       if (GOOD (word))
-	fprintf (output_file, "DDT from %o to %o\n",
+	fprintf (output_file, "DDT from %llo to %llo\n",
 		 word & 0777777, (word >> 18) & 0777777);
 
       word = get_word_at (memory, JBHRL);

--- a/ipak.c
+++ b/ipak.c
@@ -1,4 +1,4 @@
-/* Copyright (C) 2019 Lars Brinkhoff <lars@nocrew.org>
+/* Copyright (C) 2019, 2022 Lars Brinkhoff <lars@nocrew.org>
 
     This program is free software: you can redistribute it and/or modify
     it under the terms of the GNU General Public License as published by
@@ -160,6 +160,11 @@ main (int argc, char **argv)
       sixbit_to_ascii (buffer[1] ^ UMASK, string);
       fprintf (stderr, "User: %s\n", string);
       i = 5;
+    }
+  else
+    {
+      fprintf (stderr, "Unknown IPAK format.\n");
+      exit (1);
     }
 
   fprintf (stderr, "\nFile name       Words  Timestamp\n");

--- a/palx.c
+++ b/palx.c
@@ -1,4 +1,4 @@
-/* Copyright (C) 2018 Lars Brinkhoff <lars@nocrew.org>
+/* Copyright (C) 2018, 2022 Lars Brinkhoff <lars@nocrew.org>
 
     This program is free software: you can redistribute it and/or modify
     it under the terms of the GNU General Public License as published by
@@ -79,6 +79,8 @@ static void image_out (FILE *in, FILE *out, int first, int address, int count)
 {
   int i, c;
   
+  (void)first;
+
   count -= 6;
 
   if (count == 0)
@@ -162,6 +164,8 @@ symtab (FILE *in, FILE *out)
 static void
 no_symtab (FILE *in, FILE *out)
 {
+  (void)in;
+  (void)out;
   exit (0);
 }
 

--- a/pdp10-opc.c
+++ b/pdp10-opc.c
@@ -393,7 +393,7 @@ const struct pdp10_instruction pdp10_instruction[] =
   { "ashc",	0244,	PDP10_BASIC,	PDP10_ALL, 0, 0 },
   { "rotc",	0245,	PDP10_BASIC,	PDP10_ALL, 0, 0 },
   { "lshc",	0246,	PDP10_BASIC,	PDP10_ALL, 0, 0 },
-  { "fix",	0247,	PDP10_BASIC,	PDP10_KA10_SAIL },
+  { "fix",	0247,	PDP10_BASIC,	PDP10_KA10_SAIL, 0, 0 },
   { "exch",	0250,	PDP10_BASIC,	PDP10_ALL, 0, 0 },
   { "blt",	0251,	PDP10_BASIC,	PDP10_ALL, 0, 0 },
   { "aobjp",	0252,	PDP10_BASIC,	PDP10_ALL, 0, 0 },

